src/vmspawn/vmspawn.c

```diff
@@ -390,7 +674,67 @@ static int run_virtual_machine(void) {
                 _exit(EXIT_FAILURE);
         }
 
-        return wait_for_terminate_and_check(qemu_binary, child_pid, WAIT_LOG);
+#ifdef ARCHITECTURE_SUPPORTS_SMBIOS
+        int exit_status = INT_MAX;
+        if (use_vsock) {
+                r = setup_notify_parent(event, vsock_fd, &exit_status, &notify_event_source);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to setup event loop to handle vsock notify events: %m");
+        }
+#endif
+
+        /* shutdown qemu when we are shutdown */
+        (void) sd_event_add_signal(event, NULL, SIGINT, on_orderly_shutdown, PID_TO_PTR(child_pid));
+        (void) sd_event_add_signal(event, NULL, SIGTERM, on_orderly_shutdown, PID_TO_PTR(child_pid));
+
+        (void) sd_event_add_signal(event, NULL, SIGRTMIN+18, sigrtmin18_handler, NULL);
+
+        /* Exit when the child exits */
+        (void) sd_event_add_child(event, NULL, child_pid, WEXITED, on_child_exit, NULL);
+
+        r = sd_event_loop(event);
+        if (r < 0)
+                return log_error_errno(r, "Failed to run event loop: %m");
+
+#ifdef ARCHITECTURE_SUPPORTS_SMBIOS
+        if (use_vsock) {
+                if (exit_status == INT_MAX) {
+                        log_debug("Couldn't retrieve inner EXIT_STATUS from vsock");
+                        return EXIT_SUCCESS;
+                }
+                if (exit_status != 0)
+                        log_warning("Non-zero exit code received: %d", exit_status);
+                return exit_status;
+        }
+#endif
+
+        return 0;
+}
+
+static int determine_names(void) {
+        int r;
+
+        if (!arg_image)
+                return log_error_errno(SYNTHETIC_ERRNO(-EINVAL), "Missing required argument -i/--image, quitting");
```

Comment:
we try to suffix switches that take parameters in all our log messages/docs with `=`. Hence please write `--image=` rather than just `--image` here